### ISSUE TITLE
Cant Record (Dxtory Stuck At "INIT")

### ISSUE THREAD

**Issue by Orvis25** [jackun/TestAMFVFW#6]
So maybe i am misunderstanding, but this is the new codec you are making to replace the openVFW due to some issue with amd dropping support on CCC 15.7+? 

Anyway, if this is THANK YOU... But.... currently i cant get it working. 

Got it successfully installed. shows up in dxtory as "testAMFVFW64 Encoder", and when i go in game dxtory FPS counter shows up. Unfortunately when i hit start record for file output, dxtory goes orange and displays the "INIT" message, but never outputes a file. If i minimize and switch to losless, then start it, it starts recording normally and outputs a file. 

Got a fix?

Edit: 
PS: i installed AMD Media SDK kit, and the  2013 runtime. Using Dxtory V 2.0.133

**Comment by jackun** [comment]
If you don't have Fury or tonga-based R9-285/380 then make sure recording resolution is max 1920x1080 px.

**Comment by Orvis25** [comment]
My monitor is only 1920x1080p.

I am using an r9 280

**Comment by jackun** [comment]
Hmm, if you installed x64 version of vc++2013 runtime and configuration dialog works then try with logging. I haven't added much error logging, but IIRC with some more fatal errors it should show a message box. Log file is currently written to "working dir", usually somewhere in game's folder.

**Comment by Orvis25** [comment]
I enabled Log in the codec, launched multiple games, tried recording, got the orange "INIT" issue, closed, and went to "my games" documents folder.

I do not see anything named working DIR, so maybe it is not registered as an error?

ps: using windows 10, so idk what "game's folder" you are referring to other than the "my games" folder in the documents folder.

**Comment by jackun** [comment]
I mean game's install folder. Codec's configuration dialog shows up, right?

**Comment by Orvis25** [comment]
I went to the games (multiple ones) main directories, i do not see anything called " working dir". 

Yes, when dxtory is open, if i hit the config button, the box shows up giving me the options for profile, level preset, ect.

**Comment by jackun** [comment]
Ok, then it atleast loads. 

>  I went to the games (multiple ones) main directories, i do not see anything called " working dir". 

Sorry, not working dir literally, that's why there's quotes around it. It is the folder that application starts [executing in](https://en.wikipedia.org/wiki/Working_directory). Look for `TestAMFVFW.log`. 

Other than resolution, one thing that can cause a "error" is pixel bitdepth, but dxtory should use supported 32bit RGB by default. Sometimes drivers themself go wacko. Cleaning them with [DDU](http://www.wagnardmobile.com/forums/viewforum.php?f=5) or amd uninstall utility and reinstalling might help. If there's nothing useful in log then you might try that.

**Comment by Orvis25** [comment]
Found it. I i opened the file for Fallout 4 and a few others. Here is what they all say

"
Compression query: 32 0 1920x1080
CompressQuery OK
Compression query: 32 0 1920x1080
CompressQuery OK
CompressEnd
CompressEnd
"

Edit: i do not know if this matters. but i am using Cromson drivers. Does this not work with that driver set? i know it is pretty new.

**Comment by jackun** [comment]
Ok, tried myself and got that yellow init. Had to change audio device to onboard from monitor's Displayport audio device.

Try again or see other logs. There should be more info about HW capabilities and other properties like my R9 290:

> Compression query: 32 0 1680x1050
> CompressQuery OK
> Compression query: 32 0 1680x1050
> CompressQuery OK
> Capabilities:
>  Accel type: HARDWARE
>  Max bitrate: 100000000
>  Levels: 10 11 12 13 20 21 22 30 31 32 40 41 42 50 51 
>  Profiles: 66 77 100 
>  Input mem types:
>    DX11, native: Yes
>    OPENCL, native: No
>    OPENGL, native: No
>    HOST, native: No
>  Width min/max: 64/1920
>  Height min/max: 64/1920

Otherwise idk, I'll see if there's to add some more logging.

**Comment by Orvis25** [comment]
So is this an issue with dxtory, or your codec just needs an update? 

Because other than my Logitec G430 headset (and the AMD audio from the video card HDMI), i have no other spreakers / audio devices.

**Comment by jackun** [comment]
There's one weird place it might exit to `CompressEnd` and why it is twice in the log if two encode threads use the same codec instance. Haven't seen it myself but let me upload new version and try that.

**Comment by jackun** [comment]
Ok, redownload and see if there's a line of `CompressBegin: already began compressing, exiting...` in log.

**Comment by Orvis25** [comment]
Re-downloaded, installed. I deleted the old "testamdvfw" log file and booted game, hit the record button, stuck at "Init", exited, and i got the Exact same thing i posted above (so i wont re-post). 

No new info, and changing the audio dose not help.

**Comment by jackun** [comment]
Can you try converting something in virtualdub for example? It should show a messagebox on errors though they too are a bit vague.

**Comment by Orvis25** [comment]
Jackun you got steam? this may be a little easier on a MSG service. If so what is your steam name, i will add you, then edit your msg and remove if you want. 

Edit: i also have no idea how to use virtualdub. So you would have to explain how to (prefer over steam)

**Comment by jackun** [comment]
Yes, blademunky, deleting this message so google bots won't spider it :-P

**Comment by jackun** [comment]
Ok, Got emailed previous post?

**Comment by Orvis25** [comment]
I got it, but no one with that name exits acording to steam. Here add me. 

"SpartanWarrior". My icon is a picture of a anime wolf.

**Comment by jackun** [comment]
Oh right, it searches by current profile name.

**Comment by jackun** [comment]
Convolution matrix.
